pkgs/applications/networking/pjsip/default.nix

```diff
@@ -33,19 +36,35 @@ stdenv.mkDerivation rec {
     })
   ];
 
+  nativeBuildInputs =
+    lib.optional pythonSupport [ swig ];
+
   buildInputs = [ openssl libsamplerate ]
+    ++ lib.optional pythonSupport python
     ++ lib.optional stdenv.isLinux alsa-lib
     ++ lib.optional stdenv.isDarwin AppKit;
 
   preConfigure = ''
     export LD=$CC
   '';
 
+  postBuild = lib.optionalString pythonSupport ''
+    make -C pjsip-apps/src/swig/python
+  '';
+
+  outputs = [ "out" ] ++ lib.optionals pythonSupport [ "py" ];
+
+  configureFlags = [ "--enable-shared" ];
+
   postInstall = ''
     mkdir -p $out/bin
     cp pjsip-apps/bin/pjsua-* $out/bin/pjsua
     mkdir -p $out/share/${pname}-${version}/samples
     cp pjsip-apps/bin/samples/*/* $out/share/${pname}-${version}/samples
+  '' + lib.optionalString pythonSupport ''
+    (cd pjsip-apps/src/swig/python && \
+      python setup.py install --prefix=$py
```

Comment:
I think we're supposed to use `${python.interpreter}` instead of `python` as a command do it can handle odd cases like pypy.

Reply:
Ah, I've changed it to use the interpreter, though in the Makefile it still does call `python setup.py` as well. I guess that should be fine for now.